modules/apigee_edge_teams/src/Routing/TeamContextSwitcherRouteAlterSubscriber.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_teams\Routing;
+
+use Drupal\apigee_edge_teams\TeamContextManagerInterface;
+use Drupal\Core\Routing\RouteSubscriberBase;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Adds the _apigee_team_route option to developer (user) routes.
+ *
+ * @see \Drupal\apigee_edge_teams\TeamContextManager::getCorrespondingRouteNameForEntity()
+ */
+final class TeamContextSwitcherRouteAlterSubscriber extends RouteSubscriberBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function alterRoutes(RouteCollection $collection) {
+    foreach ($collection as $id => $route) {
+      // Add a corresponding team route if the team route defines a
+      // corresponding developer route.
+      if (($developer_route_id = $route->getOption(TeamContextManagerInterface::DEVELOPER_ROUTE_OPTION_NAME)) && ($developer_route = $collection->get($developer_route_id)) && empty($developer_route->getOption(TeamContextManagerInterface::TEAM_ROUTE_OPTION_NAME))) {
```

Comment:
We did consider implementing a simple lookup table but the table would have to be updated and a hook would have to be created. Then that would all have to be cached.

The route option is already very well cached and it's exposed in the routing file so it is clear that a team route relates to a developer route in some way.

Maybe this could be more clear.

> // Add a corresponding team route if the team route defines a
> // corresponding developer route.

The context switcher is a teams feature and there is a 1-to-1 relationship between  a team route and a developer route and so it doesn't we shouldn't have to specify the developer route has a corresponding team route. This adds the corresponding team route to developer routes so they are easily accessible from the route object. Without this, we would have to reverse lookup the related team route from the developer route at run-time.

Reply:
I wrote this because I also considered that what you described in the comment above and I saw better options for doing this. As you have already merged this PR and I stop thinking about how this can be done differently.